controllers/azurecluster_controller.go

```diff
@@ -215,6 +217,23 @@ func (r *AzureClusterReconciler) reconcileNormal(ctx context.Context, clusterSco
 		Port: clusterScope.APIServerPort(),
 	}
 
+	if azureCluster.Spec.IdentityRef != nil {
+		identity, err := GetClusterIdentityFromRef(ctx, clusterScope.Client, azureCluster.Namespace, azureCluster.Spec.IdentityRef)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		if identity.Namespace == azureCluster.Namespace {
+			patchhelper, err := patch.NewHelper(identity, r.Client)
+			if err != nil {
+				return reconcile.Result{}, errors.Wrap(err, "failed to init patch helper")
+			}
+			identity.ObjectMeta.OwnerReferences = azureCluster.GetOwnerReferences()
```

Comment:
I'm wondering if this would be better placed in the cluster scope so that upon scope.Close() the patch will be ensured.

Reply:
ok, makes sense, will move it

Reply:
so the patch helper takes the azurecluster when its created, so it will not work with patching the identity